source/material_model/rheology/friction_options.cc

```diff
@@ -0,0 +1,495 @@
+                }
+              catch (...)
+                {
+                  std::cerr << "FunctionParser failed to parse\n"
+                            << "\t a function\n"
+                            << "with expression \n"
+                            << "\t' " << prm.get("Function expression") << "'";
+                  throw;
+                }
+            }
+            prm.leave_subsection();
+          }
+        else if (  prm.get("Rate and state parameters b") == "None" )
+          a_and_b_source = None;
+        else
+          {
+            AssertThrow(false, ExcMessage("Unknown method for Rate and state parameters b."));
+          }
+      }
+
+
+      template <int dim>
+      double
+      FrictionOptions<dim>::
+      compute_dependent_friction_angle(const double current_edot_ii,
+                                       const unsigned int j,  // volume fraction
+                                       const std::vector<double> &composition,
+                                       typename DoFHandler<dim>::active_cell_iterator current_cell,
+                                       double current_friction,
+                                       const Point<dim> &position) const
+      {
+
+        switch (friction_dependence_mechanism)
+          {
+            case independent:
+            {
+              /* I guess here I need to say that current friction is simply the internal angle of friction. If not it will return 0.0 as its value?  */
+              break;
+            }
+            case dynamic_friction:
+            {
+              // The dynamic characteristic strain rate is used to see if dynamic or static angle of internal friction should be used.
+              // This is done as in the material_model dynamic_friction which is based on equation 13 in van Dinther et al., (2013, JGR).
+              // const double mu  = mu_d[i] + (mu_s[i] - mu_d[i]) / ( (1 + strain_rate_dev_inv2/reference_strain_rate) );
+              // which is the following using the variables in this material_model. Although here the dynamic friction coefficient
+              // is directly specified. The coefficient of friction is the tangent of the internal angle of friction.
+              // Furthermore a smoothness coefficient is added, which influences if the friction vs strain rate curve is rather
+              // step-like or more gradual.
+              const double mu = effective_friction_factor[j] * std::tan(dynamic_angles_of_internal_friction[j])
+                                + (std::tan(current_friction) - std::tan(dynamic_angles_of_internal_friction[j]))
+                                / (1. + std::pow((current_edot_ii / dynamic_characteristic_strain_rate),
+                                                 dynamic_friction_smoothness_exponent));
+              current_friction = std::atan (mu);
+              break;
+            }
+            case state_dependent_friction:
```

Comment:
a and b are called `rate and state dependent parameters`, but they are used in the `state_dependent_friction` option only. Can you rethink the names?